Update the porting guide for a new ansible version

abadger opened the pull request:
##### SUMMARY
Update porting guide for the 3.1.0 release.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Docs Pull Request


##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

abadger committed:
Update the porting guide for a new ansible version

samccann commented on docs/docsite/rst/porting_guides/porting_guide_3.rst:
```diff
@@ -14,11 +14,69 @@ Ansible 3 Porting Guide
 Ansible 3 is based on Ansible-Base 2.10, which is the same major release as Ansible 2.10.  Therefore, there is no section on ansible-base in this porting guide.  If you are upgrading from Ansible 2.9, please first consult the Ansible 2.10 porting guide before continuing with the Ansible 3 porting guide.
 
-We suggest you read this page along with the `Ansible Changelog for 3.0 <https://github.com/ansible-community/ansible-build-data/blob/main/3/CHANGELOG-v3.rst>`_ to understand what updates you may need to make.
+Porting Guide for v3.1.0
```
Why are we removing the link to the changelog here? Seems that line should stay...

abadger replied:
Thanks for catching this.  I know why it happened... I built this using a PR branch but the PR hadn't been rebased against the current main branch in a while so it didn't contain this line in the porting guide "template".  I'll fix it manually in this PR.

abadger committed:
Re-add the link to the changelog.

The changelog link was omitted because I built this using a PR that
didn't have the latest changes applied in its branch.
